test/flows/vat_payment_deadlines_flow_test.rb

```diff
@@ -0,0 +1,85 @@
+require "test_helper"
+require "support/flow_test_helper"
+
+class VatPaymentDeadlinesFlowTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup do
+    WebMock.stub_request(:get, WorkingDays::BANK_HOLIDAYS_URL)
+      .to_return(body: File.open(fixture_file("bank_holidays.json")))
+    testing_flow VatPaymentDeadlinesFlow
+  end
+
+  should "render a start page" do
+    assert_rendered_start_page
+  end
+
+  context "question: when_does_your_vat_accounting_period_end?" do
+    setup { testing_node :when_does_your_vat_accounting_period_end? }
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    context "next_node" do
+      should "have a next node of how_do_you_want_to_pay? for a date of '2013-04-30' response" do
+        assert_next_node :how_do_you_want_to_pay?, for_response: "2013-04-30"
+      end
+
+      context "validation" do
+        should "be invalid if non end-of-month date" do
+          assert_invalid_response "2013-05-15"
+        end
+
+        should "be valid if a leap year is given" do
+          assert_next_node :how_do_you_want_to_pay?, for_response: "2012-02-29"
```

Comment:
```suggestion
          assert_valid_response "2012-02-29"
```

Reply:
Following this - and the previous - suggestion this block now looks a bit odd...

```ruby
    context "next_node" do
      should "have a next node of :how_do_you_want_to_pay? for a valid response" do
        assert_valid_response: "2013-04-30"
      end
    end
```

We have a `context "next_node"` that is no longer asserting on the next node. The block looks like a validation. Are you happy this change is what you want?